tests/integration/test_task.py

```diff
@@ -80,3 +82,219 @@ def _run(self, input_files):
     print(order)
     assert len(order) == 12
     assert order == sorted(order)
+
+
+@pytest.mark.parametrize('executables', ['ls', 'mv'])
+def test_which(executables):
+    """Test the which wrapper."""
+    assert which(executables).split(os.sep)[-1] in ['ls', 'mv']
+
+
+def test_py2ncl():
+    """Test for _py2ncl func."""
+    ncl_text = _py2ncl(None, 'tas')
+    assert ncl_text == 'tas = _Missing'
+    ncl_text = _py2ncl('cow', 'tas')
+    assert ncl_text == 'tas = "cow"'
+    ncl_text = _py2ncl([1, 2], 'tas')
+    assert ncl_text == 'tas = (/1, 2/)'
+    ncl_text = _py2ncl({'cow': 22}, 'tas')
+    assert ncl_text == 'tas = True\ntas@cow = 22\n'
+    with pytest.raises(ValueError) as ex_err:
+        _py2ncl([1, "cow"], 'tas')
+    assert 'NCL array cannot be mixed type:' in str(ex_err.value)
+    with pytest.raises(ValueError) as ex_err:
+        _py2ncl({"cow": 22}, None)
+    assert 'NCL does not support nested dicts:' in str(ex_err.value)
+
+
+def test_base_task():
+    """Test BaseTask basic attributes."""
+    tasks = set()
```

Comment:
Remove the `tasks` set, it does not seem to add anything to this test, but it is confusing to the reader